unstructured_ingest/v2/processes/connectors/delta_table.py

```diff
@@ -94,21 +94,34 @@ class DeltaTableUploader(Uploader):
     connection_config: DeltaTableConnectionConfig
     connector_type: str = CONNECTOR_TYPE
 
-    @requires_dependencies(["s3fs", "fsspec"], extras="s3")
+    @requires_dependencies(["boto3"], extras="delta-table")
     def precheck(self):
         secrets = self.connection_config.access_config.get_secret_value()
         if (
             self.connection_config.aws_region
             and secrets.aws_access_key_id
             and secrets.aws_secret_access_key
         ):
-            from fsspec import get_filesystem_class
+            from urllib.parse import urlparse
+
+            from boto3 import client
 
             try:
-                fs = get_filesystem_class("s3")(
-                    key=secrets.aws_access_key_id, secret=secrets.aws_secret_access_key
+                url = urlparse(self.connection_config.table_uri)
+                bucket_name = url.netloc
+                dir_path = url.path.lstrip("/")
+
+                s3_client = client(
+                    "s3",
+                    aws_access_key_id=secrets.aws_access_key_id,
+                    aws_secret_access_key=secrets.aws_secret_access_key,
                 )
-                fs.write_bytes(path=self.connection_config.table_uri, value=b"")
+                s3_client.put_object(Bucket=bucket_name, Key=dir_path, Body=b"")
+
+                response = s3_client.get_bucket_location(Bucket=bucket_name)
+                assert self.connection_config.aws_region == response.get(
```

Comment:
This should be raising a source connection error with this as the message. Look at any of the other prechecks. `assert` should be isolated to pytests.

Reply:
I believe you wanted to say destination connection error. I changed it so it is raising `ValueError` in the try-except, which is then caught by except and raised as `DestinationConnectionError`. Check if this implementation meets your expectations or should I raise `DestinationConnectionError` already in the try-except